chromium_src/chrome/browser/prefs/pref_service_incognito_allowlist.cc

```diff
@@ -0,0 +1,30 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/prefs/persistant_pref_names.h"
+#include "build/build_config.h"
+#include "chrome/common/pref_names.h"
+
+#define GetIncognitoPersistentPrefsAllowlist \
+  GetIncognitoPersistentPrefsAllowlist_ChromiumImpl
+
+#include "src/chrome/browser/prefs/pref_service_incognito_allowlist.cc"
+
+#undef GetIncognitoPersistentPrefsAllowlist
+
+namespace prefs {
+
+std::vector<const char*> GetIncognitoPersistentPrefsAllowlist() {
+  std::vector<const char*> allowlist =
+      GetIncognitoPersistentPrefsAllowlist_ChromiumImpl();
+#if !BUILDFLAG(IS_ANDROID)
```

Comment:
`!IS_ANDROID` shouldn't be here.

Reply:
`std::size` complains when `kBravePersistentPrefNames` is empty array.

Reply:
return it as `const std::vector<>` from .cc file and use `.size()` method instead.
I think by moving it into `.cc` we should not use `std::size` anyways.

Reply:
Removed this guard - https://github.com/brave/brave-core/pull/16224/commits/89bdc6a27e43a8aa9bc4b6afeea859e8743fc239